### ISSUE TITLE
Relation manager attach action duplicate callback breaks recursive relationships

### ISSUE THREAD

**Issue by Quadrubo** [laravel-filament/filament#8067]
### Package

filament/tables

### Package Version

v3.0.25

### Laravel Version

v10.19.0

### Livewire Version

v3.0.0-beta.9

### PHP Version

PHP 8.2.9

### Problem description

I have a recursive relationship on my model. `App\Models\Task` can have many `App\Model\Task` and vice versa. When trying to attach on a relation manager, the duplicate check doesn't work correctly. I can attach as many models as I like.

Tables:
```php
         Schema::create('task_edges', function (Blueprint $table) {
            $table->foreignId('parent_id')->references('id')->on('tasks');
            $table->foreignId('child_id')->references('id')->on('tasks');

            $table->unique(['parent_id', 'child_id']);
        });
```

Relationships:
```php
    public function childTasks(): BelongsToMany
    {
        return $this->belongsToMany(Task::class, 'task_edges', 'parent_id', 'child_id');
    }

    public function parentTasks(): BelongsToMany
    {
        return $this->belongsToMany(Task::class, 'task_edges', 'child_id', 'parent_id');
    }
```

Problematic Code `AttachAction.php:264`
```php
...
fn (Builder $query): Builder => $query->where(
    $table->getRelationship()->getParent()->getQualifiedKeyName(),
    $table->getRelationship()->getParent()->getKey(),
),
...
```

Generated SQL:

```sql
SELECT DISTINCT
    `tasks`.*
FROM
    `tasks`
LEFT JOIN `task_edges` ON `tasks`.`id` = `task_edges`.`child_id`
WHERE NOT
    EXISTS(
    SELECT
        *
    FROM
        `tasks` AS `laravel_reserved_0`
    INNER JOIN `task_edges` ON `laravel_reserved_0`.`id` = `task_edges`.`parent_id`
    WHERE
        `tasks`.`id` = `task_edges`.`child_id` AND `tasks`.`id` = ?
)
ORDER BY
    `tasks`.`name` ASC;
```

Because the tables in the query have the same name (because it is the same table), the query fails as the callback adds the sql condition `AND tasks.id = ?` to the query which excludes all entries.

This is fixed by using `getKeyName()` instead of `getQualifiedKeyName()` in which case just the condition `AND id = ?` is added which correctly uses the current scope `laravel_reserved_0` and the query succeeds. 

```sql
SELECT DISTINCT
    `tasks`.*
FROM
    `tasks`
LEFT JOIN `task_edges` ON `tasks`.`id` = `task_edges`.`child_id`
WHERE NOT
    EXISTS(
    SELECT
        *
    FROM
        `tasks` AS `laravel_reserved_0`
    INNER JOIN `task_edges` ON `laravel_reserved_0`.`id` = `task_edges`.`parent_id`
    WHERE
        `tasks`.`id` = `task_edges`.`child_id` AND `id` = ?
)
ORDER BY
    `tasks`.`name` ASC;
```

### Expected behavior

The query should use the local scope incase both tables of the relationship are the same table. I don't know if this breaks other behavior.

### Steps to reproduce

1) Create a model that relates to itself:

```php
Schema::create('tasks', function (Blueprint $table) {
            $table->id();

            $table->string('name');
           
            $table->timestamps();
        });

        Schema::create('task_edges', function (Blueprint $table) {
            $table->foreignId('parent_id')->references('id')->on('tasks');
            $table->foreignId('child_id')->references('id')->on('tasks');

            $table->unique(['parent_id', 'child_id']);
        });
```

```php
public function childTasks(): BelongsToMany
    {
        return $this->belongsToMany(Task::class, 'task_edges', 'parent_id', 'child_id');
    }

    public function parentTasks(): BelongsToMany
    {
        return $this->belongsToMany(Task::class, 'task_edges', 'child_id', 'parent_id');
    }
```

2) Create a relation manager for that relationship:

```php
class ChildrenRelationManager extends RelationManager
{
    protected static string $relationship = 'childTasks';

    protected static ?string $inverseRelationship = 'parentTasks';
    ...
}
```

3) Try the attach action.

**Comment by sergejostir** [comment]
I also have the same problem https://github.com/filamentphp/filament/discussions/8950

**Comment by danharrin** [comment]
Fixed by #10078

**Comment by samgdev99** [comment]
I was facing an issue with recursive relationships despite this was already fixed by @danharrin as he mentioned before. So maybe this will help someone else in the future:

After troubleshooting the issue, I found that there are **two critical requirements** for the `AttachAction` to work properly with self-referencing `belongsToMany` relationships:

### 1. **Define Both Forward and Inverse Relationships in the Model**

Make sure your model has both relationships properly defined:

```php
// Forward relationship
public function habilitadas() : BelongsToMany
{
    return $this->belongsToMany(
        Dependencia::class,
        'dependencia_dependencia',
        'parent_id',
        'child_id'
    );
}

// Inverse relationship - THIS IS CRUCIAL
public function habilitadasPor() : BelongsToMany
{
    return $this->belongsToMany(
        Dependencia::class,
        'dependencia_dependencia',
        'child_id',
        'parent_id'
    );
}
```

### 2. **Specify the Inverse Relationship in the RelationManager**

In your RelationManager, you MUST define the inverse relationship:

```php
protected static string $relationship = 'habilitadas';
protected static ?string $inverseRelationship = 'habilitadasPor'; // THIS IS REQUIRED
```

### 3. **Use Simple String for recordTitleAttribute (Not a Closure)**

The `recordTitleAttribute` must be a simple string referencing a model attribute, not a closure/callback:

```php
// ✅ CORRECT - Simple string
->recordTitleAttribute('nombre')

// ❌ WRONG - Closure/callback won't work with AttachAction
->recordTitleAttribute(fn (Dependencia $record): string => "{$record->codigo} {$record->nombre}")
```

### **Why This Happens:**

- Filament's `AttachAction` uses the inverse relationship to automatically filter out already-attached records
- Without the inverse relationship, Filament tries to guess or use incorrect relationship names, causing SQL errors
- The `recordTitleAttribute` must be a simple string so Filament can automatically use that attribute for the select options in the attach modal

### **Final Working RelationManager:**

```php
class HabilitadasRelationManager extends RelationManager
{
    protected static string $relationship = 'habilitadas';
    protected static ?string $inverseRelationship = 'habilitadasPor'; // Critical!

    public function table(Table $table): Table
    {
        return $table
            ->recordTitleAttribute('nombre') // Simple string, not closure
            ->columns([
                TextColumn::make('codigo')->searchable()->sortable(),
                TextColumn::make('nombre')->searchable()->sortable(),
                IconColumn::make('habilitado')->boolean(),
            ])
            ->headerActions([
                Tables\Actions\AttachAction::make()
                    ->preloadRecordSelect()
                    ->multiple()
                    ->attachAnother(false)
                    ->modalWidth(MaxWidth::ThreeExtraLarge),
                // ... other actions
            ]);
    }
}
```

This solution resolved the issue where `AttachAction` was showing repeated generic labels instead of actual record names, and eliminated the SQL errors related to undefined relationship methods.
